pom.xml

```diff
@@ -750,7 +750,6 @@
         <groupId>org.apache.logging.log4j</groupId>
         <artifactId>log4j-slf4j-impl</artifactId>
         <version>${log4j.version}</version>
-        <scope>${hadoop.deps.scope}</scope>
```

Comment:
Please see: https://github.com/apache/spark/pull/35811#discussion_r824331185

Reply:
Good to know it was discussed, though not sure it was known some of the default logging behavior wouldn't work without it?

Reply:
Well, none of Spark works without Hadoop libs anyway, right?

Reply:
? Not saying logging doesn't work without Hadoop included. Just saying log4j/reload4j ends up being used under the hood so the default log4j2.properties doesn't take effect and logging doesn't work as expected

Reply:
Alternative might be also including the old log4j 1 default config file? Not a log4j expert (or even novice) by any means

Reply:
I see, I guess I mean, the idea is you must provide Hadoop no matter what. The premise here seems to be, make it work without a Hadoop distro, or maybe I misundersatnd

Reply:
Yes that is the case, and I don't know what the best way to handle that is. Is it deterministic which one gets picked so having the warning is better than logging not working as expected? Or might it still sometimes pick the other implementation? That's why I mentioned the potential alternative of just keeping the old log4j1 configuration around too (assuming that fixes the issue?)

Reply:
Could we add the impl only in the 'hadoop-provided' profile? would that make sense?

Reply:
Well it's definitely needed without that profile, the question is _also_ including it with the hadoop-provided profile (which means just removing the scope so it's provided with or without the hadoop-provided profile.)

Reply:
I see, the real problem is Hadoop has log4j-sl4j-impl 1.x and we need 2.x or am I still off base?

Reply:
Yes (or slf-reload4j with newer Hadoop releases). I think the built-in log4j2.properties files are ignored because log4j1 is running under the hood